.github/workflows/build-extra.yml

```
@@ -30,15 +30,15 @@ jobs:
   build-clang:
     runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@@ec3a7ce113134d7a93b817d10a8272cb61118579
```

It's unclear which version a SHA represents.

How about putting each intended version in a comment?

```suggestion
    # v2
    - uses: actions/checkout@@ec3a7ce113134d7a93b817d10a8272cb61118579
```

Or maybe:

```suggestion
    # actions/checkout@v2
    - uses: actions/checkout@@ec3a7ce113134d7a93b817d10a8272cb61118579
```

---

Would the comment be updated by Dependabot? Otherwise it will bitrot.

---

@topimiettinen [on Dec 14](https://github.com/netblue30/firejail/pull/4774#discussion_r769078738):

> Would the comment be updated by Dependabot? Otherwise it will bitrot.

With the current configuration, does Dependabot put something like "update
foo@v1 to foo@v2" in the commit message (I know that it can do that, but I'm
not sure if it's the default)?  If the equivalent tags can be discovered
locally/without having to search for them on GitHub, I think that would be good
enough.

If not, instead of a comment on each item, to avoid bitrot and duplication I
think we could have a mapping on a separate file.  It would only need updating
when adding a completely new "action@version" combination (i.e.: one that was
never part of any committed workflow file).  Kind of like a "package.lock"
file.  Example:

```
actions/checkout@v2: ec3a7ce113134d7a93b817d10a8272cb61118579
actions/checkout@v3: 1231231231231231231231231231231231231231
github/codeql-action/init@v1: e095058bfa09de8070f94e98f5dc059531bc6235
```

By the way, is Dependabot smart enough to update a SHA to a new SHA or would it
open a PR to bump `actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579`
to `actions/checkout@v3` in this case?

---

> @topimiettinen [on Dec 14](https://github.com/netblue30/firejail/pull/4774#discussion_r769078738):
> 
> > Would the comment be updated by Dependabot? Otherwise it will bitrot.
> 
> With the current configuration, does Dependabot put something like "update foo@v1 to foo@v2" in the commit message (I know that it can do that, but I'm not sure if it's the default)? If the equivalent tags can be discovered locally/without having to search for them on GitHub, I think that would be good enough.

Yes, here's an example commit/PR with the same settings: https://github.com/topimiettinen/libaslrmalloc/commit/0aeb6639ebfda800ace6cd36306d0869e7ce207e

> 
> If not, instead of a comment on each item, to avoid bitrot and duplication I think we could have a mapping on a separate file. It would only need updating when adding a completely new "action@version" combination (i.e.: one that was never part of any committed workflow file). Kind of like a "package.lock" file. Example:
> 
> ```
> actions/checkout@v2: ec3a7ce113134d7a93b817d10a8272cb61118579
> actions/checkout@v3: 1231231231231231231231231231231231231231
> github/codeql-action/init@v1: e095058bfa09de8070f94e98f5dc059531bc6235
> ```

For Python, there's `requirements.txt` with similar syntax (https://github.com/topimiettinen/libaslrmalloc/blob/master/.github/workflows/requirements.txt) but I don't know if that would work for GH actions. Example of Dependabot updating it: https://github.com/topimiettinen/libaslrmalloc/commit/248495bd96599b59771bd9edcd4e6e510c4e97db

> By the way, is Dependabot smart enough to update a SHA to a new SHA or would it open a PR to bump `actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579` to `actions/checkout@v3` in this case?

Only SHA is updated. There's also a GH action to verify that PRs don't change SHAs to semvers: https://github.com/zgosalvez/github-actions-ensure-sha-pinned-actions but I don't know if there's an action to verify that it actually does what it says it does and so forth.

---

Sounds good then; thanks for the examples.